### ISSUE TITLE
Internal server error is thrown after we go to the Beneficiary History Page

### ISSUE THREAD

**Issue by abaranowska1** [benetech/ServiceNet#1686]
**Bug description**
Internal server error is thrown after we go to the Beneficiary History Page
error: 
{type: "https://www.jhipster.tech/problem/problemwithmessage", title: "Internal Server Error",…}
detail: "Unable to find org.benetech.servicenet.domain.Location with id 38170397-221e-4dda-af6d-47e355900067; nested exception is javax.persistence.EntityNotFoundException: Unable to find org.benetech.servicenet.domain.Location with id 38170397-221e-4dda-af6d-47e355900067"
message: "error.http.500"
path: "/api/referrals/search"
status: 500
title: "Internal Server Error"
type: "https://www.jhipster.tech/problem/problemwithmessage"

![image](https://user-images.githubusercontent.com/61964176/106725239-7181c180-6609-11eb-86a0-b57122f182b0.png)

**To Reproduce**
1. Log in as provider user: provider
2. Go to the Beneficiary History

**Actual result**

- Internal server error is thrown after we go to the Beneficiary History Page

**Expected behavior**

- Beneficiary History page should display

**Verified on:**
- Ubuntu 20.04
- Google Chrome Browser Version 85.0.4183.83

Dev Signoffs
- [ ] Unit test added for the core business features at the minimum
- [ ] Code review passed
- [ ] CI passed successfully
- [ ] CodeQL scan for vulnerabilities in code has passed 
- [ ] Docker images have been built and deployed to the dev server

QA Signoffs
- [x] Passes Acceptance Criteria
- [x] All bug reports fixed/merged/tested

Product Signoff
- [x] Approved for release

**Comment by jkondrat** [comment]
The root cause was fixed recently (#1675) but it seems that there were a couple of referrals with non-existent locations in the dev database.
Fixed it by running the following query (no need to add a migration as #1675  wasn't released yet):
```
UPDATE referral SET to_location_id = NULL WHERE referral.to_location_id IS NOT NULL AND NOT EXISTS (SELECT FROM location l WHERE l.id = referral.to_location_id);
UPDATE referral SET from_location_id = NULL WHERE referral.from_location_id IS NOT NULL AND NOT EXISTS (SELECT FROM location l WHERE l.id = referral.from_location_id);
```

**Comment by abaranowska1** [comment]
**Tested with success**
Internal Server error is not thrown. 
![image](https://user-images.githubusercontent.com/61964176/106744420-042d5b00-6620-11eb-8119-615d942ecefc.png)
